util/log/log.go

```diff
@@ -34,74 +43,173 @@ func FatalOnPanic() {
 	}
 }
 
+func logKV(buf *bytes.Buffer, kvs ...[]interface{}) {
+	var i int
+	var s string
+	var kv []interface{}
+	for kvi := range kvs {
+		kv = kvs[kvi]
+		l := len(kv)
+		for i = 0; i < l; i++ {
+			if i%2 == 0 {
+				if kvi != 0 || i != 0 {
+					buf.WriteString(" ")
+				}
+			} else {
+				buf.WriteString("=")
+			}
+			switch kt := kv[i].(type) {
+			case string:
+				s = kt
+			case fmt.Stringer:
+				s = kt.String()
+			default:
+				s = fmt.Sprintf("%v", kt)
+			}
+			if i%2 == 0 {
+				buf.WriteString(s)
+			} else {
+				buf.WriteString(strconv.Quote(s))
+			}
+		}
+		if i%2 != 0 {
+			panic(fmt.Sprintf("odd number of key-value pairs passed: %v", kv))
+		}
+	}
+	buf.WriteString("\n")
+}
+
+func kvPrint(kvs ...[]interface{}) func(*bytes.Buffer) {
```

Comment:
Are you intending to use this in more than one place?  As it is this is just confusing things, I would just inline it in logDepth: 

```
clog.PrintWith(sev, depth+1, func(buf *bytes.Buffer{
  logKv(buf, headerKV(sev, depth+1, msg), contextKV(ctx), kvs)
})